src/Microsoft.AspNet.Mvc.Core/DefaultControllerActionArgumentBinder.cs

```diff
@@ -20,15 +22,18 @@ public class DefaultControllerActionArgumentBinder : IControllerActionArgumentBi
         private readonly IModelMetadataProvider _modelMetadataProvider;
         private readonly MvcOptions _options;
         private readonly IObjectModelValidator _validator;
+        private readonly ICompositeMetadataDetailsProvider _compositeDetailsProvider;
```

Comment:
This can't be here. This is a layering violation. I'll read further and come up with a suggestion.